Introduce CassandraSessionConfigurator (#23899)
trinodb/trino: docs/src/main/sphinx/connector/cassandra.md

| `cassandra.consistency-level`    | Consistency levels in Cassandra refer to the level of consistency to be used for both read and write operations.  More information about consistency levels can be found in the [Cassandra consistency] documentation. This property defaults to a consistency level of `ONE`. Possible values include `ALL`, `EACH_QUORUM`, `QUORUM`, `LOCAL_QUORUM`, `ONE`, `TWO`, `THREE`, `LOCAL_ONE`, `ANY`, `SERIAL`, `LOCAL_SERIAL`. |
| `cassandra.allow-drop-table`     | Enables {doc}`/sql/drop-table` operations. Defaults to `false`.                                                                                                                                                                                                                                                                                                                                                             |

> | `cassandra.security`             | Should be set to `PASSWORD` for password based authentication. Defaults to `NONE`.                                                                                                                                                                                                                                                                                                                                          |

Reviewer: Maybe use

```
Configure authentication to Cassandra. Defaults to `NONE`. Set to `PASSWORD` for basic authentication, and configure `cassandra.username` and `cassandra.password`.
```

Also maybe we should redo the two following ones to mention that `cassandra.security=PASSWORD` is required?  

See my following suggestion.

> | `cassandra.username`             | Username used for authentication to the Cassandra cluster. This is a global setting used for all connections, regardless of the user connected to Trino.                                                                                                                                                                                                                                                                    |

Reviewer: ```suggestion
| `cassandra.username`             | Username for basic authentication to the Cassandra cluster. Requires `cassandra.security=PASSWORD`. |
```

> | `cassandra.password`             | Password used for authentication to the Cassandra cluster. This is a global setting used for all connections, regardless of the user connected to Trino.                                                                                                                                                                                                                                                                    |

Reviewer: ```suggestion
| `cassandra.password`             | Password for basic authentication to the Cassandra cluster. Requires `cassandra.security=PASSWORD`.                                                                                                                                                                                                                                                                    |
```
